### PR TITLE
Add SimpleCheck

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
SimpleCheck will pass if either both StyleCheck and FastTest passed or pull request is labeled 'allow simple fail'

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

closes #38471

### CODE REVIEW

**Comment by yakov-olkhovskiy** [comment]
@Felixoid I'm not sure how it will behave on adding/removing label though - I don't see on.pull_request.types contains [labeled,
unlabeled]

**Comment by alexey-milovidov** [comment]
@yakov-olkhovskiy we have many pull requests where neither Fast Test nor Style Check are run.

It should be implemented in that way: https://github.com/ClickHouse/ClickHouse/issues/18507:
make SimpleCheck green at the start and turn it red if Fast Test or Style Check failed.

Otherwise, it cannot be enabled on GitHub.

**Comment by yakov-olkhovskiy** [comment]
@alexey-milovidov my solution is to add complementary check for ignored paths - your proposed solution is impossible I think.
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

**Comment by alexey-milovidov** [comment]
Ok.

**Comment by alexey-milovidov** [comment]
The approach to mark some checks as green and then update is possible if you are not using GitHub Actions and pushing updates to GitHub API directly. We used this approach before.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -179,6 +179,17 @@ jobs:
           # shellcheck disable=SC2046
           docker rm -f $(docker ps -a -q) ||:
           sudo rm -fr "$TEMP_PATH" "$CACHES_PATH"
+  SimpleCheck:
+    needs: [StyleCheck, FastTest]
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    steps:
+      - name: Allow simple fail
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'allow simple fail') }}
```
Comment (alexey-milovidov) [inline .github/workflows/pull_request.yml:188]: The label should be named in a more neutral way.
Not "allow fail" but "skip check".

File: .github/workflows/pr-pass.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+on:
+  pull_request:
+    types:
+      - synchronize
+      - reopened
+      - opened
+    branches:
+      - master
+    # Сomplementary paths
+    paths:
```
Comment (Felixoid) [inline .github/workflows/pr-pass.yml:12]: no, it must not be any particular path. The check should work for everything

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -179,6 +179,17 @@ jobs:
           # shellcheck disable=SC2046
           docker rm -f $(docker ps -a -q) ||:
           sudo rm -fr "$TEMP_PATH" "$CACHES_PATH"
+  SimpleCheck:
```
Comment (Felixoid) [inline .github/workflows/pull_request.yml:182]: I don't get it. The plan is to create a commit status, like our other statuses, see the screenshot 
image

You'll slip into hell with solutions based on GH actions. Better add a status to something global like https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/run_check.py, and then to the checks we want blocking merge, like https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/fast_test_check.py and https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/build_check.py adding overwriting for the same status. See examples for https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/fast_test_check.py#L17

**Review by Felixoid — CHANGES_REQUESTED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**